src/app/beer_garden/app.py

```diff
@@ -82,11 +82,17 @@ def initialize(self):
         ]
 
         # Only want to run the MongoPruner if it would do anything
+        cancel_after = config._DB_SPEC["items"]["ttl"]["items"]["in_progress"].get(
```

Comment:
Meant to comment on this last time and forget to apparently.  The way you want to interact with the config is something like:

```python
cancel_after = config.get("db.ttl.in_progress")
```

You don't want to reference the specs directly and also what you're doing here seems to be getting the default value, as opposed to the value that the user actually set.

Also, I think we talked about at some point the fact that the entire "db.ttl" config section is retrieved immediately after this, which means you could slightly refactor so that you get that once and then pull the "in_progress" part of it for use with `cancel_after`.

Reply:
No, something is not setup right there in however you are trying to do this.  You will pretty much never get `None` back for those values because there are defaults defined.  So even if there was nothing in your custom config, the `get` would return the defaults.

In any case, it works something like this:

```python
In [2]: ttl_config = beer_garden.config.get("db.ttl")

In [3]: ttl_config
Out[3]: <Box: {'action': -1, 'info': 15, 'in_progress': 1, 'file': 15}>

In [4]: ttl_config.in_progress
Out[4]: 1
```